src/modules/EventModule.cpp

```diff
@@ -0,0 +1,463 @@
+		{
+			QStringList args = message.content().split(' ');
+
+			Permissions::ContainsPermission(client, channel.guildId(), message.author().id(), CommandPermission::ADMIN, [this, args, message, channel](bool result)
+				{
+					GuildSetting* setting = &GuildSettings::GetGuildSetting(channel.guildId());
+					if (!result)
+					{
+						UmikoBot::Instance().createMessage(message.channelId(), "**You don't have permissions to use this command.**");
+						return;
+					}
+
+					if (args.size() < 2) {
+						UmikoBot::Instance().createMessage(channel.id(), "**Wrong Usage of Command!**");
+						return;
+					}
+
+					snowflake_t guild = channel.guildId();
+					snowflake_t chan = channel.id();
+
+					auto& roles = message.mentionRoles();
+
+					if (roles.isEmpty())
+					{
+						UmikoBot::Instance().createMessage(channel.id(), "**Wrong Usage of Command!\n**Please mention some role(s) so that I can do the needed.");
+						return;
+					}
+
+					for (auto& role : roles)
+					{
+						m_EventWhitelist[guild].removeOne(role);
+					}
+
+					UmikoBot::Instance().createMessage(channel.id(), "**The roles have been removed.**\nPeople with the role(s) can no longer launch or end events!");
+
+				});
+		});
+	RegisterCommand(Commands::EVENT_LAUNCH, "launch", [this](Discord::Client& client, const Discord::Message& message, const Discord::Channel& channel)
+	{
+		QStringList args = message.content().split(' ');
+
+		client.getGuildMember(channel.guildId(), message.author().id())
+			.then([=](const Discord::GuildMember& member)
+			{
+					bool found = false;
+
+					for (auto& allowedRole : m_EventWhitelist[channel.guildId()])
+					{
+						if (member.roles().contains(allowedRole))
+						{
+							found = true;
+							break;
+						}
+					}
+
+					if (!found)
+					{
+						UmikoBot::Instance().createMessage(channel.id(), "**You don't have permissions to launch events.**");
+						return;
+					}
+
+					auto& config = getEventServerData(channel.guildId());
+					CurrencyModule* currencyModule = static_cast<CurrencyModule*>(UmikoBot::Instance().GetModuleByName("currency"));
+					auto& currencyConfig = currencyModule->getServerData(channel.guildId());
+
+					if (args.size() != 3)
+					{
+						UmikoBot::Instance().createMessage(message.channelId(), "**Wrong Usage of Command!** ");
+						return;
+					}
+
+					if (config.isEventRunning)
+					{
+						UmikoBot::Instance().createMessage(message.channelId(), "An event is already running. Please stop that to start a new event!");
+						return;
+					}
+
+					double time = args.at(2).toDouble();
+
+					QRegExp re("[+]?\\d*\\.?\\d+");
+					if (!re.exactMatch(args.at(2)))
+					{
+						UmikoBot::Instance().createMessage(message.channelId(), "**Wrong Usage of Command!**.");
+						return;
+					}
+					if (time > 24)
+					{
+						UmikoBot::Instance().createMessage(message.channelId(), "**Currently you can't launch events with an expiry time of more than one day**.");
```

Comment:
Didn't we fix this?

Reply:
Nope, @Sid200 didn't open a PR, he fixed that! 😄